src/storage/src/hummock/local_version.rs

```diff
@@ -236,12 +231,10 @@ impl LocalVersion {
     ) -> Vec<HummockEpoch> {
         let new_max_committed_epoch = new_pinned_version.max_committed_epoch;
         if self.pinned_version.max_committed_epoch() < new_max_committed_epoch {
-            self.replicated_batches
-                .retain(|epoch, _| *epoch > new_pinned_version.max_committed_epoch);
             assert!(self
                 .shared_buffer
                 .iter()
-                .all(|(epoch, _)| *epoch > new_pinned_version.max_committed_epoch));
+                .all(|(epoch, _)| *epoch >= new_pinned_version.max_committed_epoch));
```

Comment:
We remove sharedbuffer with epoch le committed_epoch during the sync, so we should keep >.

Reply:
Fix it